javascript-typing-tutor/main.js

```diff
@@ -1,14 +1,42 @@
 var $char = document.querySelectorAll('span');
 var i = 0;
+var success = 0;
+var fail = 0;
+var $success = document.querySelector('.success');
+var $fail = document.querySelector('.fail');
+var $accuracy = document.querySelector('.accuracy');
+var $popUp = document.querySelector('#popUp');
+var $SObutton = document.querySelector('.SObutton');
 
 function check(event) {
-  if (event.key === $char.item(i).textContent) {
-    $char.item(i).className = 'passed';
-    $char.item(i + 1).className = 'underlined';
+  if (event.key === $char[i].textContent) {
+    $char[i].className = 'passed';
+    $char[i + 1].className = 'underlined';
+    success += 1;
     i += 1;
   } else {
-    $char.item(i).className = 'error';
+    $char[i].className = 'error';
+    fail += 1;
   }
 }
 
+function showPopUp(event) {
+  if (event.key === $char[i].textContent && i === $char.length - 1) {
+    $success.textContent += success;
+    $fail.textContent += fail;
+    var accuracy = (success / (success + fail) * 100).toFixed(2) + '%';
+    $accuracy.textContent += accuracy;
+    $popUp.className = 'row justify-center align-center fixed blackBG';
+
+  }
+}
+
+function startOver(event) {
+  success = 0;
+  fail = 0;
+  window.location.reload();
```

Comment:
Oh nooooooo 😱

`location.reload` is a forbidden tool. It completely reloads the page, which is an **_extremely_** intensive process for the browser. Instead, you should reset all your elements to the state they would be in when the page loads using javascript, and reset the value of any variables in JavaScript to what their initial value would be as well.